### PR TITLE
feat: make certain fields nullable to match NOMIS nullability

### PR DESCRIPTION
**Pull request by ReedSoftware**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/Referral.kt [hunk 1]
```diff
@@ -53,9 +53,9 @@ data class Referral(
 
   @Column val releaseDate: LocalDate? = null,
 
-  @Column(nullable = false) val descriptionOfConcern: String,
+  @Column(nullable = false) val descriptionOfConcern: String?,
 
-  @Column(nullable = false) val knownReasons: String,
+  @Column(nullable = false) val knownReasons: String?,
```
Comment (bill-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/Referral.kt:58]: are we okay to keep nullable=false annotation for these now-nullable columns?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/service/CsipRecordService.kt [hunk 2]
```diff
@@ -41,8 +41,14 @@ class CsipRecordService(
...
-    val incidentInvolvement =
-      referenceDataRepository.getReferenceData(INCIDENT_INVOLVEMENT, request.referral.incidentInvolvementCode)
+    val incidentInvolvement = if (request.referral.incidentInvolvementCode != null) {
+      referenceDataRepository.getReferenceData(
+        INCIDENT_INVOLVEMENT,
+        request.referral.incidentInvolvementCode,
+      )
+    } else {
+      null
+    }
```
Comment (bill-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/service/CsipRecordService.kt:51]: can convert to one-liner
```
 val incidentInvolvement = request.referral.incidentInvolvementCode?.let { referenceDataRepository.getReferenceData(INCIDENT_INVOLVEMENT, it)  }
```

File: src/main/resources/migration/common/V8__ParentChildIndexes.sql [hunk 3]
```diff
@@ -15,8 +15,8 @@ CREATE INDEX idx_investigation_record_id ON investigation(record_id);
 CREATE INDEX idx_interview_record_id ON interview(record_id);
 CREATE INDEX idx_interview_interviewee_role_id ON interview(interviewee_role_id);
 
-CREATE INDEX idx_decisions_and_actions_decision_outcome_id ON decisions_and_actions(decision_outcome_id);
-CREATE INDEX idx_decisions_and_actions_decision_outcome_signed_off_role_id ON decisions_and_actions(decision_outcome_signed_off_by_role_id);
+CREATE INDEX idx_decision_and_actions_decision_outcome_id ON decision_and_actions(decision_outcome_id);
+CREATE INDEX idx_decisions_and_actions_decision_outcome_signed_off_role_id ON decision_and_actions(decision_outcome_signed_off_by_role_id);
```
Comment (bill-moj) [inline src/main/resources/migration/common/V8__ParentChildIndexes.sql:20]: `idx_decisions_and_actions_decision_outcome_signed_off_role_id`

do we want to change decisions to decision in this for consistency?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/Referral.kt (hunk 1 above)
Comment (ReedSoftware) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/Referral.kt:58]: Good spot, thanks.

File: src/main/resources/migration/common/V8__ParentChildIndexes.sql (hunk 3 above)
Comment (ReedSoftware) [reply src/main/resources/migration/common/V8__ParentChildIndexes.sql:20]: Yep, thought I'd done that - thanks

**Review by bill-moj — APPROVED** [review]

**Pull request merged**